test/cmd/builds.sh

```diff
@@ -18,10 +18,35 @@ os::cmd::expect_success 'oc new-build openshift/ruby-20-centos7 https://github.c
 os::cmd::expect_success 'oc get bc/ruby-hello-world'
 os::cmd::expect_success 'cat "${OS_ROOT}/Dockerfile" | oc new-build -D - --name=test'
 os::cmd::expect_success 'oc get bc/test'
+
+template='{{with .spec.output.to}}{{.kind}} {{.name}}{{end}}'
+
+# Build from Dockerfile with output to ImageStreamTag
 os::cmd::expect_success "oc new-build --dockerfile=\$'FROM centos:7\nRUN yum install -y httpd'"
-os::cmd::expect_success 'oc get bc/centos'
-os::cmd::expect_success 'oc delete all --all'
+os::cmd::expect_success_and_text "oc get bc/centos --template '${template}'" '^ImageStreamTag centos:latest$'
+
+# Build from Dockerfile with output to DockerImage
+os::cmd::expect_success "oc new-build -D \$'FROM openshift/origin:v1.1' --to-docker"
+os::cmd::expect_success_and_text "oc get bc/origin --template '${template}'" '^DockerImage library/origin:latest$'
+
+# Build from Dockerfile with given output ImageStreamTag spec
+os::cmd::expect_success 'oc delete is/origin'
```

Comment:
If the `oc delete` lines are cleanup from the previous test, I'd stick them there instead of them leading the next test. Otherwise these LGTM as far as formatting and readability, can't comment on correctness.